feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/data/repository/NftRepositoryImpl.kt

```diff
@@ -64,45 +73,86 @@ class NftRepositoryImpl(
+            subscriptionBuilder,
+            nftLocal
+        ).onStart {
+            subscriptionBuilder.subscribe(coroutineContext)
         }
+            .map { address -> address to nftLocal }
     }
 
     override suspend fun getLocalNft(nftIdentifier: String): NftLocal {
         return nftDao.getNft(nftIdentifier)
     }
 
+    override suspend fun getLocalNfts(nftIdentifiers: List<String>): List<NftLocal> {
+        return nftDao.getNfts(nftIdentifiers)
+    }
+
     override suspend fun initialNftSync(
         metaAccount: MetaAccount,
-        forceOverwrite: Boolean,
+        forceOverwrite: Boolean
     ): Unit = withContext(Dispatchers.IO) {
         val chains = chainRegistry.currentChains.first()
 
         val syncJobs = chains.flatMap { chain ->
+            initialNftSyncForChain(chain, metaAccount, forceOverwrite, skipFirstBlock = false)
+        }
+
+        syncJobs.joinAll()
+    }
+
+    override suspend fun initialNftSyncForChainId(
+        chainId: ChainId,
+        metaAccount: MetaAccount,
+        forceOverwrite: Boolean,
+        skipFirstBlock: Boolean
+    ): List<Job> {
+        val chain = chainRegistry.getChain(chainId)
+        return initialNftSyncForChain(chain, metaAccount, forceOverwrite, skipFirstBlock)
+    }
+
+    private suspend fun initialNftSyncForChain(
+        chain: Chain,
+        metaAccount: MetaAccount,
+        forceOverwrite: Boolean,
+        skipFirstBlock: Boolean
+    ): List<Job> {
+        return coroutineScope {
+            var blockHash: BlockHash? = null
+            if (skipFirstBlock) {
+                blockHash = getNextBlockHash(chainId = chain.id)
+            }
             nftProvidersRegistry.get(chain).map { nftProvider ->
                 // launch separate job per each nftProvider
                 launch {
                     // prevent whole sync from failing if some particular provider fails
                     runCatching {
-                        nftProvider.initialNftsSync(chain, metaAccount, forceOverwrite)
+                        nftProvider.initialNftsSync(chain, metaAccount, forceOverwrite, at = blockHash)
                     }.onFailure {
                         Log.e(NFT_TAG, "Failed to sync nfts in ${chain.name} using ${nftProvider::class.simpleName}", it)
                     }
                 }
             }
         }
+    }
 
-        syncJobs.joinAll()
+    private suspend fun getNextBlockHash(chainId: ChainId): String {
+        return coroutineScope {
+            val subscriptionBuilder = storageSharedRequestsBuilderFactory.create(chainId)
+            val runtime = chainRegistry.getRuntime(chainId)
+            val storage = runtime.metadata.system().storage("Number")
+            val blockNumberKey = storage.storageKey()
+            val blocksFlow = subscriptionBuilder.subscribe(blockNumberKey)
+            subscriptionBuilder.subscribe(this)
+            blocksFlow.drop(1).first().block
```

Comment:
I think its better to check that value of system.number is bigger by 1 from previous since due to chainHead reorgs next block number update may not yeld next block

Reply:
Done